process/elasticproc/tokens/sovereignIndexTokensHandler.go

```diff
@@ -0,0 +1,166 @@
+package tokens
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/elastic/go-elasticsearch/v7"
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/data/esdt"
+
+	"github.com/multiversx/mx-chain-es-indexer-go/client"
+	"github.com/multiversx/mx-chain-es-indexer-go/client/logging"
+	"github.com/multiversx/mx-chain-es-indexer-go/data"
+	indexerdata "github.com/multiversx/mx-chain-es-indexer-go/process/dataindexer"
+	"github.com/multiversx/mx-chain-es-indexer-go/process/elasticproc"
+	"github.com/multiversx/mx-chain-es-indexer-go/process/elasticproc/converters"
+	"github.com/multiversx/mx-chain-es-indexer-go/process/elasticproc/factory"
+)
+
+type sovereignIndexTokensHandler struct {
+	indexingEnabled        bool
+	mainChainElasticClient elasticproc.DatabaseClientHandler
+	esdtPrefix             string
+}
+
+// NewSovereignIndexTokensHandler creates a new sovereign index tokens handler
+func NewSovereignIndexTokensHandler(mainChainElastic factory.ElasticConfig, esdtPrefix string) (*sovereignIndexTokensHandler, error) {
+	var mainChainElasticClient elasticproc.DatabaseClientHandler
+	if mainChainElastic.Enabled {
```

Comment:
1. Why are you creating `NewElasticClient` inside your constructor instead of receiving it from constructor?
2. Your code COULD panic if `mainChainElastic.Enabled = false` and someone calls `mainChainElasticClient`, since you will have a nil instance(it should be disabled if not enabled)

Reply:
Changed as suggested